pkg/cli/cluster_free_disk_space.go

```diff
@@ -0,0 +1,125 @@
+		Use:          "free-disk-space",
+		Short:        "Returns the amount of free disk space (in bytes) for a given storage class",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			classes, err := clientSet.StorageV1().StorageClasses().List(cmd.Context(), metav1.ListOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to list cluster storage classes: %w", err)
+			}
+
+			var selectedClass *v1.StorageClass
+			for _, class := range classes.Items {
+				val, ok := class.Annotations[IsDefaultStorageClassAnnotation]
+				if ok && val == "true" && forStorageClass == "" {
+					selectedClass = &class
+					logger.Printf("Selected default storage class %q", selectedClass.Name)
+					break
+				}
+
+				if class.Name == forStorageClass {
+					selectedClass = &class
+					logger.Printf("Selected storage class %q", selectedClass.Name)
+					break
+				}
+			}
+
+			if selectedClass == nil {
+				if forStorageClass == "" {
+					return fmt.Errorf("failed to find default storage class")
+				}
+				return fmt.Errorf("failed to find storage class %q", forStorageClass)
+			}
+
+			switch selectedClass.Provisioner {
+			case OpenEBSLocalProvisioner:
+				freeSpaceGetter, err := clusterspace.NewOpenEBSFreeDiskSpaceGetter(clientSet, logger, openEBSImage, selectedClass.Name)
+				if err != nil {
+					fmt.Print(logbuf.String())
+					return fmt.Errorf("failed to start openebs free space getter: %w", err)
+				}
+
+				volumes, err := freeSpaceGetter.OpenEBSVolumes(cmd.Context())
+				if err != nil {
+					fmt.Print(logbuf.String())
+					return fmt.Errorf("failed to get openebs free space: %w", err)
+				}
+
+				for node, volume := range volumes {
+					if openEBSNode == "" {
+						fmt.Printf("%d\t%s\n", volume.Free, node)
+						continue
+					}
+
+					if node != openEBSNode {
+						logger.Printf("Skipping node %q", node)
+						continue
+					}
+
+					fmt.Printf("%d\n", volume.Free)
+					return nil
+				}
+
+				if forStorageClass != "" {
+					return fmt.Errorf("failed to collect openebs free space: node %q not found", openEBSNode)
+				}
+				return nil
+
+			case RookCephFSProvisioner, RookRBDProvisioner:
+				freeSpaceGetter, err := clusterspace.NewRookFreeDiskSpaceGetter(clientSet, rookClientSet, selectedClass.Name)
+				if err != nil {
+					return fmt.Errorf("failed to start rook free space getter: %w", err)
+				}
+
+				space, err := freeSpaceGetter.GetFreeSpace(cmd.Context())
+				if err != nil {
+					return fmt.Errorf("failed to get rook free space: %w", err)
+				}
+
+				fmt.Println(space)
+				return nil
+
+			default:
+				return fmt.Errorf("provisioner %q is not supported", selectedClass.Provisioner)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&forStorageClass, "storageclass", "", "The storage class for which to check the free disk space.")
+	cmd.Flags().StringVar(&openEBSImage, "openebs-image", "eeacms/rsync:2.3", "The image used by OpenEBS disk free evaluation pod.")
```

Comment:
Why the value is `eeacms/rsync:2.3`?
What happens if we inform an invalid option?
Should we not validate the inputs? 

Also, can we run the command without inform this flags? If not, then we need mark then as required, see `MarkFlagRequired("name")`

Reply:
`eeacms/rsync:2.3` is the default image used by `pvmigrate`, that is why it is here. if an invalid input is provided the command will fail and an error will be reported back. all flags are optional.

Reply:
If we change the version would still be that the default value?
Could we create a const on top and set the default value with a comment on top for help in the future if we need to understand it?